` dragonfly/snapshot_test.py::test_s3_snapshot[df_factory0]` EOF during read-response on client

```
            # save to S3 + flush + load from S3 (with local --dir)
>           await async_client.execute_command("SAVE", "DF", s3_path, "snapshot")
```

error during client read response

```
        if isinstance(response, ResponseError):
>           raise response from None
E           redis.exceptions.ResponseError: end of stream
```

https://github.com/dragonflydb/dragonfly/actions/runs/25628455376/job/75227645646#step:7:7913

--- a/src/server/db_slice.cc
+++ b/src/server/db_slice.cc
@@ -1378,7 +1378,10 @@ void DbSlice::FlushChangeToEarlierCallbacks(DbIndex db_ind, Iterator it, uint64_
     // does not change during the serialization, therefore we allow at most one serializer
     // reading the bucket at the same time.
     if (bucket_version <= cb_version) {
-      ccb->second(db_ind, ChangeReq{it.GetInnerIt()});
+      // Key might have been deleted because another FlushChangeToEarlierCallbacks finished
+      // and allowed a mutation to pass while this one was suspended
+      if (auto inner_it = it.GetInnerIt(); !inner_it.is_done())
+        ccb->second(db_ind, ChangeReq{inner_it});
     }
     ++ccb;
   }

